Tests/Tests/CDTURLSessionTests.m

```diff
@@ -55,7 +103,55 @@ - (void)testRequestInterceptorsSetCorrectly
 
     XCTAssertEqual(task.requestInterceptors.count, 1);
     XCTAssertEqual(task.requestInterceptors[0], countingInterceptor);
+    XCTAssertEqual(task.responseInterceptors.count, 1);
+    XCTAssertEqual(task.responseInterceptors[0], countingInterceptor);
     XCTAssertEqual(NSURLSessionTaskStateSuspended, task.state);
 }
 
+- (void)testInterceptorsGetCalled
+{
+    CDTCountingHTTPRequestInterceptor *countingInterceptor =
+        [[CDTCountingHTTPRequestInterceptor alloc] init];
+    CDTURLSession *session = [[CDTURLSession alloc] initWithDelegate:nil
+                                                      callbackThread:[NSThread currentThread]
+                                                 requestInterceptors:@[ countingInterceptor ]];
+
+    NSURLRequest *request =
+        [NSURLRequest requestWithURL:[NSURL URLWithString:@"http://localhost:5984"]];
+
+    CDTURLSessionTask *task = [session dataTaskWithRequest:request completionHandler:nil];
+
+    [task resume];
+
+    while (task.state != NSURLSessionTaskStateCompleted) {
+        [NSThread sleepForTimeInterval:0.1f];
+    }
+
+    XCTAssertEqual(countingInterceptor.timesRequestIntercepted, 1);
+    XCTAssertEqual(countingInterceptor.timesResponseIntercepted, 1);
+}
+
+- (void)testInterceptorsCanRetryRequests
+{
+    CDTRetryingHTTPInterceptor *replayingInterceptor = [[CDTRetryingHTTPInterceptor alloc] init];
+    CDTURLSession *session = [[CDTURLSession alloc] initWithDelegate:nil
+                                                      callbackThread:[NSThread currentThread]
+                                                 requestInterceptors:@[ replayingInterceptor ]];
+
+    NSURLRequest *request =
+        [NSURLRequest requestWithURL:[NSURL URLWithString:@"http://localhost:5984"]];
+
+    CDTURLSessionTask *task = [session dataTaskWithRequest:request completionHandler:nil];
+
+    XCTAssertEqual(task.responseInterceptors.count, 1);
+    XCTAssertEqual(task.responseInterceptors[0], replayingInterceptor);
+    XCTAssertEqual(NSURLSessionTaskStateSuspended, task.state);
+
+    [task resume];
+    while (task.state != NSURLSessionTaskStateCompleted) {
+        [NSThread sleepForTimeInterval:0.1f];
+    }
+    XCTAssertEqual(replayingInterceptor.timesCalled, 2);
+}
```

Comment:
I suggest making CDTRetryingHTTPInterceptor's init method take the number of times it should set retry to true. Then you can have a further test where you make sure that if you set the number to something silly like a million, the max retries value kicks in. Otherwise you're not testing that feature.